qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy

```diff
@@ -119,34 +120,61 @@ class IntegrationsSplunkViolationsTest extends BaseSpecification {
         boolean hasNetworkViolation = false
         boolean hasProcessViolation = false
         def port = splunkDeployment.splunkPortForward.getLocalPort()
-        for (int i = 0; i < 15; i++) {
-            log.info "Attempt ${i} to get violations from Splunk"
-            def searchId = SplunkUtil.createSearch(port, "| from datamodel Alerts.Alerts")
+        withRetry(20, 5) {
+            def searchId = SplunkUtil.createSearch(port, "search sourcetype=stackrox-violations")
             TimeUnit.SECONDS.sleep(15)
             Response response = SplunkUtil.getSearchResults(port, searchId)
             // We should have at least one violation in the response
-            if (response == null) {
-                continue
-            }
+            assert response != null
             results = response.getBody().jsonPath().getList("results")
-            if (results.isEmpty()) {
-                continue
-            }
+            assert !results.isEmpty()
             hasNetworkViolation = results.any { isNetworkViolation(it) }
             hasProcessViolation = results.any { isProcessViolation(it) }
-            if (hasNetworkViolation && hasProcessViolation) {
-                log.info "Success!"
-                break
-            }
+            log.debug "Found violations in Splunk: \n${results}"
+            log.info "Current Splunk index contains " +
+                    "any Network Violation: ${hasNetworkViolation} and any Process Violation: ${hasProcessViolation}"
+            assert hasNetworkViolation && hasProcessViolation
+        }
+
+        log.info "Starting conversion of ACS violations to Splunk alerts"
```

Comment:
[nit] Super nitpicking but maybe add a comment or change the log to reflect that we are forcing the conversion that otherwise will be done by the cronjob.